File: GrnGame/nanobind/classe/son.py
from ...chemin import renvoie_chemin_abs

class _son:
    def __init__(self, utils):
        self.utils = utils
        self.jeu = self.utils.jeu
    #francais
    def jouer(self, lien, boucle=0, canal=-1,volume = 64):
        g = self.utils._g
        if not g:
            return
        lien_abs = renvoie_chemin_abs(lien)
        self.jeu.jouer_son(g, lien_abs, boucle, canal,volume)

    def arreter(self, lien):
        g = self.utils._g
        if not g:
            return
        lien_abs = renvoie_chemin_abs(lien)
        self.jeu.arreter_son(g, lien_abs)

    def arreter_canal(self, canal):
        self.jeu.arreter_canal(canal)

    def pause_canal(self, canal):
        self.jeu.pause_canal(canal)

    def pause(self, lien):
        g = self.utils._g
        if not g:
            return
        lien_abs = renvoie_chemin_abs(lien)
        self.jeu.pause_son(g, lien_abs)

    def reprendre_canal(self, canal):
        self.jeu.reprendre_canal(canal)

    def reprendre(self, lien):
        g = self.utils._g
        if not g:
            return
        lien_abs = renvoie_chemin_abs(lien)
        self.jeu.reprendre_son(g, lien_abs)

    #anglais
    def play(self, path, loop=0, channel=-1, volume=64):
        return self.jouer(path, loop, channel, volume)

    def stop(self, path):
        return self.arreter(path)

    def stop_channel(self, channel):
        return self.arreter_canal(channel)

    def pause_channel(self, channel):
        return self.pause_canal(channel)

    def pause_sound(self, path):
        return self.pause(path)

    def resume_channel(self, channel):
        return self.reprendre_canal(channel)

    def resume(self, path):
        return self.reprendre(path)

File: src/bindings_c/son.c
/*
 * Bindings C pour le systeme audio.
 */

#include "../moteur/son/son.h"

/* Joue un son avec options de boucle, canal et volume */
void playSound(const char *path, int loop, int channel, int volume) {
    jouer_son(path, loop, channel, volume);
}

/* Arrete un son identifie par son chemin */
void stopSound(const char *path) { arreter_son(path); }

/* Met en pause un son identifie par son chemin */
void pauseSound(const char *path) { pause_son(path); }

/* Reprend un son en pause */
void resumeSound(const char *path) { reprendre_son(path); }

/* Arrete la lecture d'un canal audio */
void stopChannel(int channel) { arreter_canal(channel); }

/* Met en pause un canal audio */
void pauseChannel(int channel) { pause_canal(channel); }

/* Reprend un canal audio en pause */
void resumeChannel(int channel) { reprendre_canal(channel); }

/* Charge tous les sons d'un dossier en memoire */
void loadSongFolder(const char *folder) { charger_tout_les_sons(folder); }

/* Libere tous les sons en memoire */
void freeSongFolder(void) { liberer_gestionnaire_son(); }

/* mettre la cle pour les sons */
void setSongKey(int index, int value) { mettre_son_cle(index, value); }

/* mettre liv pour les sons */
void setSongIv(int index, int value) { mettre_son_iv(index, value); }